samples/cpp/digits.cpp

```diff
@@ -1,375 +1,185 @@
-    "Usage:\n"
-    << argv[0] << endl;
-}
-
-static void split2d(const Mat& image, const Size cell_size, vector<Mat>& cells)
-{
-    int height = image.rows;
-    int width = image.cols;
-
-    int sx = cell_size.width;
-    int sy = cell_size.height;
+const char *keys =
+    "{ help     h  | | Print help message. }"
+    "{ modelBin    | | Path to a binary .caffemodel file contains trained network.}"
+    "{ modelTxt    | | Path to a .prototxt file contains the model definition of trained network.}"
+    "{ width       | 640 | Set the width of the camera }"
+    "{ height      | 480 | Set the height of the camera }"
+    "{ thr         | 0.8 | Confidence threshold. }";
 
-    cells.clear();
+// Find best class for the blob (i.e. class with maximal probability)
+static void getMaxClass(const Mat &probBlob, int *classId, double *classProb);
 
-    for (int i = 0; i < height; i += sy)
-    {
-        for (int j = 0; j < width; j += sx)
-        {
-            cells.push_back(image(Rect(j, i, sx, sy)));
-        }
-    }
-}
+void predictor(dnn::Net net, Mat &roi, int &class_id, double &probability);
 
-static void load_digits(const char* fn, vector<Mat>& digits, vector<int>& labels)
+int main(int argc, char **argv)
 {
-    digits.clear();
-    labels.clear();
-
-    String filename = samples::findFile(fn);
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
 
-    cout << "Loading " << filename << " ..." << endl;
-
-    Mat digits_img = imread(filename, IMREAD_GRAYSCALE);
-    split2d(digits_img, Size(SZ, SZ), digits);
-
-    for (int i = 0; i < CLASS_N; i++)
+    if (argc == 1 || parser.has("help"))
     {
-        for (size_t j = 0; j < digits.size() / CLASS_N; j++)
-        {
-            labels.push_back(i);
-        }
+        parser.printMessage();
+        return 0;
     }
-}
 
-static void deskew(const Mat& img, Mat& deskewed_img)
-{
-    Moments m = moments(img);
+    int vWidth = parser.get<int>("width");
+    int vHeight = parser.get<int>("height");
+    float confThreshold = parser.get<float>("thr");
+    String modelTxt = parser.get<String>("modelTxt");
+    String modelBin = parser.get<String>("modelBin");
 
-    if (abs(m.mu02) < 0.01)
+    dnn::Net net;
+    try
     {
-        deskewed_img = img.clone();
-        return;
+        net = dnn::readNetFromCaffe(modelTxt, modelBin);
     }
-
-    float skew = (float)(m.mu11 / m.mu02);
-    float M_vals[2][3] = {{1, skew, -0.5f * SZ * skew}, {0, 1, 0}};
-    Mat M(Size(3, 2), CV_32F);
-
-    for (int i = 0; i < M.rows; i++)
+    catch (cv::Exception &ee)
     {
-        for (int j = 0; j < M.cols; j++)
+        cerr << "Exception: " << ee.what() << endl;
+        if (net.empty())
         {
-            M.at<float>(i, j) = M_vals[i][j];
+            cout << "Can't load the network by using the flowing files:" << endl;
+            cout << "modelTxt: " << modelTxt << endl;
+            cout << "modelBin: " << modelBin << endl;
+            exit(-1);
```

Comment:
Error code should not be negative. Maybe just `return 1;`?

Reply:
fixed